packages/ics721/src/query.rs

```diff
@@ -81,20 +81,17 @@ pub trait Ics721Query {
         let token_id = TokenId::new(token_id);
         let class_id = ClassId::new(class_id);
 
-        let Some(token_metadata) = TOKEN_METADATA.may_load(
-            deps.storage,
-            (class_id.clone(), token_id.clone()),
-        )? else {
-        // Token metadata is set unconditionaly on mint. If we have no
-        // metadata entry, we have no entry for this token at all.
-        return Ok(None)
+        let Some(token_metadata) =
+            TOKEN_METADATA.may_load(deps.storage, (class_id.clone(), token_id.clone()))?
+        else {
+            // Token metadata is set unconditionaly on mint. If we have no
+            // metadata entry, we have no entry for this token at all.
+            return Ok(None);
```

Comment:
If there is no token metadata we just return a None. Does the caller see that as an error? What does Ok(None) mean to the contract that is calling this?

Reply:
ics721 has prepared for handling token metadata. prepared means it is stored in ics721 rn - since core `cw721` package does not support it per se. see zeke's comment in `callback_mint()` in execute.rs:

```rs
    fn callback_mint(
        &self,
        deps: DepsMut,
        class_id: ClassId,
        tokens: Vec<Token>,
        receiver: String,
    ) -> Result<Response<T>, ContractError> {
        let receiver = deps.api.addr_validate(&receiver)?;
        let cw721_addr = CLASS_ID_TO_NFT_CONTRACT.load(deps.storage, class_id.clone())?;

        let mint = tokens
            .into_iter()
            .map(|Token { id, uri, data }| {
                // We save token metadata here as, ideally, once cw721
                // supports on-chain metadata, this is where we will set
                // that value on the debt-voucher token. Note that this is
                // set for every token, regardless of if data is None.
                TOKEN_METADATA.save(deps.storage, (class_id.clone(), id.clone()), &data)?;
...
```

Flow is like this:
- on send from source, metadata is loaded (which is currently None)
- pass to target as part of `NonFungibleTokenPacketData`
- save metadata in ics721 on target chain

Reply:
I just want to be clear here, so ics721. is storing onchain metadata on ics721? As in,  when someone moves their nft from A to B, metadata is saved on chain B's ics721 contract. In our original discussion of this issue, we talked about querying original chain to get the onchain metadata from its original home chain. Does this mean we have abandoned that method?

Reply:
No, we didn't abandoned it. What we store here is on initial transfer and storing data provided from `NonFungibleTokenPacketData`. Also added another commit with some more comments on metadata. Pls check also my other 2 comments on TOKEN_METADATA. Please note, this is not part of this PR - only added some comments in source for clarifications.